src/Server.cpp

```diff
@@ -852,33 +950,97 @@ void handleCompleteRequestHeader(Server& server, int clientFd, Connection& conne
 	}
 
 	server.findTargetResource(connection, connection.m_request);
+	if (isCGIRequested(connection)) {
+		connection.m_request.hasCGI = true;
+		ProcessOps processOps;
+		CGIHandler cgiHandler(connection, processOps);
+		if (connection.m_request.httpStatus == StatusInternalServerError) {
+			connection.m_status = Connection::BuildResponse;
+			server.modifyEvent(clientFd, EPOLLOUT);
+			return;
+		}
+		cgiHandler.execute(connection.m_request, connection.location, processOps, server.getEpollWrapper().getEpollFd(),
+			server.getConnections());
+		connection.m_pipeToCGIWriteEnd = cgiHandler.getPipeInWriteEnd();
+		connection.m_pipeFromCGIReadEnd = cgiHandler.getPipeOutReadEnd();
+		connection.m_cgiPid = cgiHandler.getCGIPid();
+		if (!server.registerCGIFileDescriptor(connection.m_pipeToCGIWriteEnd, EPOLLOUT, connection)
+			|| !server.registerCGIFileDescriptor(connection.m_pipeFromCGIReadEnd, EPOLLIN, connection)) {
```

Comment:
Doesn't this error when pipeToCGI is -1?

If we don't have method POST, the pipe is not opened so it's still -1?

Reply:
True, adapted to only check for the first if request.method = MethodPost. Added this also in the dup2Process of the CGIHandler as this was causing an error as well. Tested it with _reqCgiGetTime_